File: pipe/pipe.py
#!/usr/bin/python

import os, sys


class Pipe:
	print "The child will write text to a pipe and "
	print "the parent will read the text written by child..."

def reverse(s):
    txt = ''
    for i in range(len(s)-1, -1, -1):
        txt += s[i]
	print txt
    return
# file descriptors r, w for reading and writing
r, w = os.pipe() 
s = raw_input("Enter text: ")
processid = os.fork()
if processid:
    # This is the parent process 
    # Closes file descriptor w
    os.close(w)
    r = os.fdopen(r)
    print "Parent reading"
    print s
    str = r.read()
    print "text =", str   
    sys.exit(0)
else:
    # This is the child process
    os.close(r)
    w = os.fdopen(w, 'w')
    print "Child writing"
    reverse(s);
    w.write("Text written by child...")
    w.close()
    print "Child closing"
    sys.exit(0)

File: pipe/src/pipe.cpp
#include <iostream>
#include <unistd.h>
#include <string.h>
using namespace std;
class P{
	int fd[2],i,j;
	char a[20],buf[1024],temp;
public:
	   void p1();
	   void p2();
};
void P::p1()
{
	cout<<"\n------This is Process A-------";
	cout<<"\nEnter the String :";
	cin>>a;
	pipe(fd);
	write(fd[1],a,strlen(a));
	read(fd[0],buf,strlen(a));
    close(pipe(fd));
}
void P::p2()
{
     i=0;
     j=strlen(a)-1;
     cout<<"\n--------This is Process B---------\n";
     if(fork()== 0)
     {
       while(i<j)
       {
    	temp=a[i];
    	a[i]=a[j];
    	a[j]=temp;
    	i++;
    	j--;
       }
      write(fd[1],a,strlen(a));
      read(fd[0],buf,strlen(a));
      close(pipe(fd));
      write(1,buf,strlen(a));
     }
     else if(fork()== -1)
     {
    	 cout<<"\n unsuccessful creation...";
     }
}
int main()
{
	P w;
	w.p1();
	w.p2();
	return 0;
}
